docs/generating-tests.rst

```diff
@@ -0,0 +1,417 @@
+transaction.  The ``expect`` field does not need to specify a state completely,
+but it should specify some features of some accounts.  ``expect`` field is a list.
+Each element talks about some elements of the multi-dimensional array defined in ``transaction`` field.
+
+.. code::
+
+   "expect" : [
+        {
+            "indexes" : {
+                "data" : 0,
+                "gas" : -1,
+                "value" : -1
+            },
+            "network" : ["Frontier", "Homestead"],
+            "result" : {
+                "095e7baea6a6c7c4c2dfeb977efac326af552d87" : {
+                    "balance" : "2000000000000000010",
+                    "storage" : {
+                        "0x" : "0x01",
+                        "0x01" : "0x01"
+                    }
+                },
+                "2adc25665018aa1fe0e6bc666dac8fc2697ff9ba" : {
+                    "balance" : "20663"
+                },
+                "a94f5374fce5edbc8e2a8697c15331677e6ebf0b" : {
+                    "balance" : "99979327",
+                    "nonce" : "1"
+                }
+            }
+        },
+        {
+            "indexes" : {
+                "data" : 1,
+                "gas" : -1,
+                "value" : -1
+            },
+        ...
+        }
+    ]
+
+
+``indexes`` field specifies a subset of the transactions.  ``-1`` means "whichever".
+``"data" : 0`` points to the first element in the ``data`` field in ``transaction``.
+
+``network`` field is somehow similar.  It specifies the versions of the protocol
+for which the expectation applies.  For expectations common to all versions, say
+``"network" : ALL``.
+
+Filling the Test
+----------------
+
+The test filler file is not for consumption.  The filler file needs to be filled
+into a test.  ``testeth`` has the ability to compute the post-state from the test filler,
+and produce the test.  The advantage of the filled test is that it can catch any post-state
+difference between clients.
+
+First, if you created a new subdirectory for the filler, you need to edit the source
+of ``cpp-ethereum`` so that ``testeth`` recognizes the new subdirectory.  The file to 
+edit is `StateTests.cpp`_, which lists the names of the subdirectories scanned for 
+GeneralStateTest filters.
+
+.. _`StateTests.cpp`: https://github.com/ethereum/aleth/blob/master/test/tools/jsontests/StateTests.cpp
+
+
+After building ``testeth``, you are ready to fill the test.
+
+.. code:: bash
+
+   ETHEREUM_TEST_PATH="<LOCAL_PATH_TO_ETH_TESTS>" test/testeth -t GeneralStateTests/stExample2 -- --filltests --checkstate
+
+
+where the environmental variable ``ETHEREUM_TEST_PATH`` should point to the directory 
+where ``tests`` repository is checked out.  ``stExample2`` should be replaced with 
+the name of the subdirectory you are working on.  ``--filltests`` option tells ``testeth``
+to fill tests.  ``--checkstate`` tells ``testeth`` to check the final states against
```

Comment:
Note to self: this part has to be updated as well depending on the answer from @winsvega above.

Reply:
Ok, have removed --checkstate here as well.